### ISSUE TITLE
Jack-in not doing anything on Windows 10 if you just open a clj file, not sure if bug

### ISSUE THREAD

**Issue by SarmBoJim** [BetterThanTomorrow/calva#444]
Hi I just installed VS Code 1.39.2 on Windows 10 1903 and have Leiningen 2.9.1, Clojure 1.10.0 and have AdoptOpenJDK 64-bit build 13+33 installed.
Then I installed the Calva extension 2.0.55

I used lein new wonderland to make a new project in my Documents folder then used Open file from the VS Code file menu to open the core.clj file in the new project. Then I used Ctrl+Alt+C and Ctrl+Alt+J to try to jack in but nothing happens. So I select "Start a project REPL and connect" from the Command Palette menu and again nothing happens.

OK so I tried adding my wonderland project folder as a workspace in VS Code and now the Jack-in works and gave me the option of choosing a project type and has successfully connected to a REPL session.

The  documentation on https://calva.readthedocs.io/en/latest/connect.html does not mention that you need to add your project folder as a workspace in VS Code for jack-in to to work so maybe this is a bug or the documentation should be updated to reflect this?

Shouldn't Jack-in just work from opening a single .clj file or is there something I am missing about how it all works? I am fairly new to Clojure and VS Code, though have done a little bit of Clojure before in Leiningen and Cursive and other environments.

**Comment by bpringe** [comment]
Hi! The jack-in command opens a "project" repl, and requires some kind of project files (project.clj, deps.edn, etc) to know what your dependencies are so it can load them.

I do think the docs could use more clarification, possibly here:
>Open a file somewhere in your project directory in VS Code.

Perhaps it could specify that the project itself needs to be open in vs code.

Though here it does allude to that, but could be clearer.
>When Jack-in starts it will depend on the project type, and whether ClojureScript is involved or not, and if it is, what kind of ClojureScript project, what will happen next. Calva will analyze the project files and will then give you prompts with selections based on what is found there.

@PEZ can fill in anything I may be missing.

**Comment by SarmBoJim** [comment]
Yes I think so because when I read that line " Open a file somewhere in
your project directory in VS Code" I read it as open a file from your
project directory in Windows Explorer from VS Code which I believed using
Open file to open the core.clj file fulfilled.

So yes I think having the project itself loaded into VS Code should be made
more explicit or noobs like me might be hitting this brickwall when
starting and maybe giving up straight away.

So do I need to contact @PEZ <https://github.com/PEZ> to see about getting
those docs updated or do you do this? Not sure.

Thanks for the help and response ;)

**Comment by PEZ** [comment]
Hello. I agree this is a bug, either in the implementation or in the documentation. I'll have a look.

Did you find this doc? https://calva.readthedocs.io/en/latest/workspace-layouts.html btw?

**Comment by PEZ** [comment]
OK. So this has changed lately. It used to be that you could open a file and jack in. I think maybe we should restore that functionality, but for now at least I fixed that instruction: https://calva.readthedocs.io/en/latest/connect.html

Thanks for reporting @SarmBoJim !

**Comment by SarmBoJim** [comment]
Hi Peter,
Thanks for the response, yes I quickly read through all the documentation
on there the day before I installed Calva but the later stuff didn't really
jump out to me at the time.
And I was just focused on getting started and jacking in so I was mostly
just looking at the first entry in the docs which dealt with that.

I just checked the page you mentioned has the fixed instruction but I am
not sure if it is the right page I am viewing and if it is then I still
don't think it is quite clear enough for a noob.
I see you can edit it on Github but I am new to that too at the moment and
unsure how to do it. So my suggestion is to take this part here:


   1. Open your project root directory in VS Code.
   2. Issue the command *Start a REPL project and connect*: ctrl+alt+c
   ctrl+alt+j.
   3. Answer the quick-pick prompts telling Calva about project types and
   what profiles to start.

And change it to something like this:


   1. Select Add folder to workspace from the VS Code File menu and select
   your project root folder/directory
   2. Issue the command *Start a REPL project and connect*: ctrl+alt+c
   ctrl+alt+j.
   3. Answer the quick-pick prompts telling Calva about project types and
   what profiles to start


Otherwise I think when I am reading the former I would be just trying to
open the project folder from the file menu.
I am new to the workspace concept too so I think the distinction that is
made more explicit in the latter version that I suggest will make that more
clear and specific to noobs like myself.

Thanks for the help and thank you very much for your work on Calva, I am
loving both it and VS Code so far and using it to get back into learning
Clojure ;)

**Comment by PEZ** [comment]
Thanks for the cheering on! We appreciate that people appreciate Calva! ❤️ 

About the documentation. Thing is that this will work:

> Otherwise I think when I am reading the former I would be just trying to open the project folder from the file menu.

It is how I do it. The workspace folder stuff is for people who need that. So I choose to just link to that information in a ”see also”.

**Comment by SarmBoJim** [comment]
Ah OK thanks for the clarification and you're welcome!
When I started out trying to learn Clojure the dev environment and tooling
stuff kind of made it a bit hard to get into, there was the full-on pro
stuff like Emacs and IntelliJ +Cursive or stuff for beginners that felt
pretty rough.
The VS Code plus Calva combo seems to me like Clojure might finally have a
great answer for an entry point into Clojure that can get stuff done as
well, so yeah, definitely appreciated ;)

**Comment by bpringe** [comment]
That's great to hear, @SarmBoJim! When starting Clojure myself, I also was not familiar with Emacs or IntelliJ, but I was very familiar with VS Code. This is the main reason I like contributing to Calva when my time permits. I want the Clojure community to grow, and I think Calva is now a key piece in the puzzle. I think @PEZ feels the same. 😄

**Comment by PEZ** [comment]
I get super happy by hearing that you think Calva serves this function, @SarmBoJim , it is precisely what we aim for: an easy to use start, and productive to stick with. Please consider pasting that text as a review https://marketplace.visualstudio.com/items?itemName=betterthantomorrow.calva#review-details ❤️

**Comment by SarmBoJim** [comment]
You're welcome, and thanks for the suggestion, I've added my Calva review
as Rowan Barnes on the Marketplace.
I hope Calva gets all the recognition it deserves ;)

**Comment by PEZ** [comment]
Great! You seem to have inspired someone, because yet another review has been issued.

**Comment by SarmBoJim** [comment]
Sweet! Hope it continues to grow, I found out about Calva through the
regular email notifications I get from Clojureverse btw; if you're at all
interested to know.
There was an article on Calva you did there and about how easy it was to
make extensions for VS Code.
I looked into it further and was amazed to see VS Code topped the Stack
Overflow developer surveys 2019 for most widely used dev tools. Pretty
incredible for something which has only been out for a few years. And I'm
guessing it'll grow a lot more with people like me just picking it up now
having just noticed it's explosion in popularity.
So I jumped in and grabbed Calva and am happy to see that it seems like
this is the kind of accessible Clojure setup I've been wishing for.
